src/main/java/communication/MessageSender.java

```diff
@@ -0,0 +1,171 @@
+package communication;
+
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+import org.apache.log4j.Logger;
+import org.zeromq.ZMQ;
+
+import control.FBase;
+import crypto.CryptoProvider.EncryptionAlgorithm;
+import de.hasenburg.fbase.model.GetMissedMessageResponse;
+import exceptions.FBaseCommunicationException;
+import exceptions.FBaseEncryptionException;
+import model.JSONable;
+import model.config.NodeConfig;
+import model.data.DataIdentifier;
+import model.data.DataRecord;
+import model.data.MessageID;
+import model.messages.Command;
+import model.messages.Envelope;
+import model.messages.Message;
+
+/**
+ * Sends requests to designated receivers.
+ * 
+ * @author jonathanhasenburg
+ *
+ */
+public class MessageSender extends AbstractSender {
```

Comment:
This should probably also have a less generic name unless you're planning to additional send methods.

Reply:
See comment MessageReceiver